cdap-common/src/main/java/io/cdap/cdap/common/metrics/MetricsReporterHook.java

```diff
@@ -102,6 +113,21 @@ public void postCall(HttpRequest request, HttpResponseStatus status, HandlerInfo
 
       // todo: report metrics broken down by status
       collector.increment("response." + name, 1/*, "status:" + code*/);
+
+      // store response time metric
+      LocalDateTime startTime = requestStarts.getIfPresent(request);
+      if (startTime != null) {
+        requestStarts.invalidate(request);
+        long responseTimeMs = ChronoUnit.MILLIS.between(startTime, LocalDateTime.now());
+        // metricNameCache avoids concatenating the metric name each time
+        String metricName = metricNameCache.get(handlerInfo.getMethodName());
```

Comment:
No need to encode the context into the metrics name. It is already part of the MetricsContext.

Reply:
Now that I understand metrics better, removed it. Thanks.